### ISSUE TITLE
[sklearn#34500] Pandas output path fails to reject infinite values during transform

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#320]
**Mirrored from [scikit-learn#34500](https://github.com/scikit-learn/scikit-learn/issues/34500)**

Original title: Pandas output path fails to reject infinite values during transform
Original labels: Sprint
Created: 2026-07-17T10:28:49Z

---

### Describe the bug and give evidence about its user-facing impact

When using `set_output(transform="pandas")`, `SelectKBest.transform` does not reject infinite values in a pandas `DataFrame`. This is a regression introduced by [PR #25102](https://github.com/scikit-learn/scikit-learn/pull/25102).

The pandas-output path passes `cast_to_ndarray=False` to `_validate_data`, which causes `check_array` to skip its `force_all_finite` validation. This allows invalid numeric inputs through `transform` instead of raising the expected validation error.

### Steps/Code to Reproduce

```python
import numpy as np
import pandas as pd
from sklearn.feature_selection import SelectKBest, f_classif

X_train = pd.DataFrame(
    {
        "a": [0.0, 1.0, 0.0, 1.0],
        "b": [1.0, 2.0, 1.0, 2.0],
    }
)
y = [0, 0, 1, 1]

selector = SelectKBest(f_classif, k=1).set_output(transform="pandas")
selector.fit(X_train, y)

X_bad = X_train.copy()
X_bad.loc[0, "a"] = np.inf

print("Input containing infinity:")
print(X_bad)
print()

try:
    result = selector.transform(X_bad)
    print("Transform succeeded (no error raised):")
    print(result)
except Exception as exc:
    print("Error:", exc)
```

### Expected Results

The input containing infinity should be rejected during `transform`:

```
Input containing infinity:
     a    b
0  inf  1.0
1  1.0  2.0
2  0.0  1.0
3  1.0  2.0

Error: Input X contains infinity or a value too large for dtype('float64').
```

(This is the behavior when `set_output` is not used.)

### Actual Results

No validation error is raised; transform silently succeeds:

```
Input containing infinity:
     a    b
0  inf  1.0
1  1.0  2.0
2  0.0  1.0
3  1.0  2.0

Transform succeeded (no error raised):
     b
0  1.0
1  2.0
2  1.0
3  2.0
```

### Versions

```
scikit-learn 1.9.0
Python 3.11.13
Linux
NumPy 2.4.6
SciPy 1.17.1
```

### Interest in fixing the bug

The pandas-output path should preserve DataFrame containers without changing the estimator's validation behavior. Infinite values should continue to be rejected consistently during `transform` regardless of the output format setting.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Addressed in upstream PR(s). See scikit-learn/scikit-learn PRs for #34500.
